### PR TITLE
Create new helm chart for Zulip

### PR DESCRIPTION
**Pull request by mrtndwrd**

### CODE REVIEW

File: kubernetes/chart/zulip/values-local.yaml.example [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - host: zulip.example.com
+      paths:
+      - path: /
+  tls:
+    - secretName: zulip-tls-secret
+      hosts:
+        - zulip.example.com
+
+memcached:
+  memcachedPassword: set-secure-password
```
Comment (varac) [inline kubernetes/chart/zulip/values-local.yaml.example]: Even though I see your good intention in the comment at the top of this file I think it's not a good practice to set default passwords, even in a `values-local.yaml.example` file. Please comment them at least so users won't make the mistake of copying this file and forgetting to change a default password.

File: kubernetes/chart/zulip/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+apiVersion: v2
+description: zulipchat
+name: zulip
+type: application
+
```
Comment (varac) [inline kubernetes/chart/zulip/Chart.yaml]: Please add an icon:

```suggestion
icon: https://raw.githubusercontent.com/zulip/zulip/main/static/images/logo/zulip-icon-square.svg
```

```
❯ helm lint             
==> Linting .
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```

File: kubernetes/chart/zulip/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+apiVersion: v2
+description: zulipchat
```
Comment (varac) [inline kubernetes/chart/zulip/Chart.yaml]:
```suggestion
description: A chat for distributed teams which combines the immediacy of real-time chat with an email threading model. With Zulip, you can catch up on important conversations while ignoring irrelevant ones. 
```

File: kubernetes/chart/zulip/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.2.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "4.7-0"
```
Comment (varac) [inline kubernetes/chart/zulip/Chart.yaml:15]: The GH release page only states `4.7`

```suggestion
appVersion: "4.7"
```

File: kubernetes/chart/zulip/Chart.yaml (hunk 2 above)
Comment (varac) [inline kubernetes/chart/zulip/Chart.yaml]:
```suggestion
sources:
  - https://github.com/zulip/zulip
  - https://github.com/zulip/docker-zulip
  - https://hub.docker.com/r/zulip/docker-zulip
```

**Review by varac — CHANGES_REQUESTED** [review]
Besides my comments, looks good!

File: kubernetes/chart/zulip/values-local.yaml.example [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Replace each occurrence of "set-secure-password" with a different random
+# password.
+# Replace each occurrence of "zulip.example.com" with the domain you want Zulip
+# to be reached on.
+
+# Supports OIDC and the below `create_realm` script
+image:
+  repository: open.greenhost.net:4567/stackspin/stackspin/zulip
```
Comment (varac) [inline kubernetes/chart/zulip/values-local.yaml.example]: I'd use the official image here.

File: kubernetes/chart/zulip/templates/NOTES.txt [hunk 6]
```diff
@@ -0,0 +1,15 @@
+1. Zulip should now be available on:
+{{- if .Values.ingress.enabled }}
+{{- range .Values.ingress.hosts }}
+  http://{{ .host }}
```
Comment (varac) [inline kubernetes/chart/zulip/templates/NOTES.txt:4]:
```suggestion
  https://{{ .host }}
```

File: kubernetes/chart/zulip/templates/NOTES.txt [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+{{- else }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "zulip.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
+  kubectl port-forward $POD_NAME 8080:80
+{{- end }}
+
+2. To create a realm so you can sign in:
+
+  kubectl -n {{ .Release.Namespace }} exec -it $(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "zulip.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}") -c zulip /home/zulip/deployments/current/manage.py generate_realm_creation_link
+
```
Comment (varac) [inline kubernetes/chart/zulip/templates/NOTES.txt:15]: I test-installed the chart, so far so good, but I couldn't succeed with creating a realm:

```
❯ kubectl -n zulip exec -it $(kubectl get pods --namespace zulip -l "app=zulip,release=zulip" -o jsonpath="{.items[0].metadata.name}") -c zulip /home/zulip/deployments/current/manage.py generate_realm_creation_link

error: error executing jsonpath "{.items[0].metadata.name}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].metadata.name}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":"", "selfLink":""}}


kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
error: arguments in resource/name form may not have more than one slash
```

**Review by varac — CHANGES_REQUESTED** [review]

File: kubernetes/chart/zulip/Chart.yaml (hunk 4 above)
Comment (mrtndwrd) [reply kubernetes/chart/zulip/Chart.yaml:15]: The [Docker tag I'm installing is `4.7-0`](https://github.com/greenhost/docker-zulip#welcome-to-docker-zulip), so I prefer to keep it like this to make that clear.

File: kubernetes/chart/zulip/templates/NOTES.txt (hunk 6 above)
Comment (mrtndwrd) [reply kubernetes/chart/zulip/templates/NOTES.txt:4]: Although I like https, `http` will always work, whereas `https` here will only work if people have added a cert-manager annotation to their ingress (or another way to enable TLS), which is not the default.

File: kubernetes/chart/zulip/Chart.yaml (hunk 4 above)
Comment (varac) [reply kubernetes/chart/zulip/Chart.yaml:15]: What about this change ? I don't see it in your last 2 commits

File: kubernetes/chart/zulip/templates/NOTES.txt (hunk 6 above)
Comment (varac) [reply kubernetes/chart/zulip/templates/NOTES.txt:4]: What about this change ?

**Review by varac — CHANGES_REQUESTED** [review]
Nice, please have a look at my two comments which were not addressed in you latest commits.

**Review by mrtndwrd — COMMENTED** [review]
forgotten comments

File: kubernetes/chart/zulip/Chart.yaml (hunk 4 above)
Comment (varac) [reply kubernetes/chart/zulip/Chart.yaml:15]: OK!

File: kubernetes/chart/zulip/templates/NOTES.txt (hunk 6 above)
Comment (varac) [reply kubernetes/chart/zulip/templates/NOTES.txt:4]: Agreed

**Comment by varac** [comment]
If you see a merge button please hit it, I don't :P

**Pull request merged**